packages/contracts/contracts/MarginlyPool.sol

```diff
@@ -704,24 +712,24 @@ contract MarginlyPool is IMarginlyPool {
       uint256 positionDiscountedBaseDebtPrev = position.discountedBaseAmount;
       uint256 realQuoteCollateral = calcRealQuoteCollateral(
         position.discountedQuoteAmount,
-        position.discountedBaseAmount
+        positionDiscountedBaseDebtPrev
       );
       uint256 realBaseDebt = baseDebtCoeff.mul(positionDiscountedBaseDebtPrev, Math.Rounding.Up);
 
       {
-        // quoteInMaximum is defined by user input limitPriceX96
-        uint256 quoteInMaximum = Math.mulDiv(limitPriceX96, realBaseDebt, FP96.Q96);
-
         realCollateralDelta = swapExactOutput(true, realQuoteCollateral, realBaseDebt, swapCalldata);
-        if (realCollateralDelta > quoteInMaximum) revert Errors.SlippageLimit();
+
+        // quoteInMaximum is defined by user input limitPriceX96
+        if (realCollateralDelta > Math.mulDiv(limitPriceX96, realBaseDebt, FP96.Q96)) revert Errors.SlippageLimit();
         swapPriceX96 = getSwapPrice(realCollateralDelta, realBaseDebt);
 
         uint256 realFeeAmount = Math.mulDiv(params.swapFee, realCollateralDelta, WHOLE_ONE);
-        chargeFee(realFeeAmount);
+        uint256 managerFee = msg.sender == positionOwner ? 0 : params.managerFee;
```

Comment:
`uint256 managerFee = msg.sender == positionOwner ? 0 : params.managerFee;` expression could be placed before if(){}else{} clause